src/main/java/org/isf/lab/service/LabIoOperations.java

```diff
@@ -104,6 +104,33 @@ public List<Laboratory> getLaboratory(String exam, LocalDateTime dateFrom, Local
 				                                                      TimeTools.truncateToSeconds(dateTo.with(LocalTime.MAX)));
 	}
 	
+	/**
+	 * Return a list of exams ({@link Laboratory}s) between specified dates and matching passed exam name
+	 * @param exam - the exam name as <code>String</code>
+	 * @param dateFrom - the lower date for the range
+	 * @param dateTo - the highest date for the range
+	 * @param patient - the object of patient 
+	 * @return the list of {@link Laboratory}s 
+	 * @throws OHServiceException
+	 */
+	public List<Laboratory> getLaboratory(String exam, LocalDateTime dateFrom, LocalDateTime dateTo, Patient patient) throws OHServiceException {
+		List<Laboratory> laboritories = new ArrayList<>();
+		
+		if(exam.equals("") && patient != null) {
+			laboritories = repository.findByLabDateBetweenAndExamDescriptionAndPatientCode(dateFrom, dateTo, exam, patient.getCode());
+		}
+		if( exam.equals("") && patient == null ) {
+			laboritories = repository.findByLabDateBetweenAndExam_DescriptionOrderByLabDateDesc(dateFrom, dateTo, exam);
+		}
+		if(patient != null && exam.equals("")) {
```

Comment:
```suggestion
		if (patient != null && exam.equals("")) {
```

Reply:
How is line 125 different than line 119 (other than the ordering of the comparisons?

Reply:
I don't know... let me ask (it was also on informatici repo the same way)